### PR TITLE
Add buildkit to workspace

### LINKED ISSUE #10: Add buildkit to the gitpod config

### PR DESCRIPTION
**Pull request by csweichel**
Needed this for my own work and it was quick enough to add.

Fixes #10

### CODE REVIEW

File: .gitpod.yml [hunk 1]
```diff
@@ -4,7 +4,19 @@ image:
...
 tasks:
+  - command: sudo buildkitd --group gitpod
+  - command: docker run -p 5000:5000 registry
+    openMode: split-right
   - init: go get -v ./...
+    openMode: tab-after
+    command: |
+      gp await-port 5000
+      go generate -v ./...
+      go run main.go build --context example $(gp url 5000 | cut -d/ -f3)/dazzle
```
Comment (rl-gitpod) [inline .gitpod.yml:15]: I'd prefer to see this in the Makefile so it is a little more standard & more easily reused.
See [my branch](https://github.com/gitpod-io/dazzle/blob/rl/upgrade-all-the-things/example/Makefile)

File: .gitpod.yml [hunk 2]
```diff
@@ -4,7 +4,19 @@ image:
...
+ports:
+  - port: 5000
```
Comment (rl-gitpod) [inline .gitpod.yml:17]: Why make it public? 
localhost:5000/dazzle doesn't require opening it up.

File: .gitpod.yml [hunk 3]
```diff
@@ -4,7 +4,19 @@ image:
 checkoutLocation: dazzle
 workspaceLocation: dazzle
 tasks:
+  - command: sudo buildkitd --group gitpod
+  - command: docker run -p 5000:5000 registry
```
Comment (rl-gitpod) [inline .gitpod.yml:8]: Use registry:2 tag instead of latest to be more explicit and less - albeit minimal - likelihood of related issues?

**Review by rl-gitpod — COMMENTED** [review]
Hrm, I need better notification - I didn't see this and implemented it as well. Not a big deal and adding the buildkit tar to the image is slightly quicker than my more naive approach as is the use of the gitpod terminal positioning/control.

File: .gitpod.yml (hunk 2 above)
Comment (csweichel) [reply .gitpod.yml:17]: Docker runs in its own network namespace. I doubt `localhost:5000` would work

File: .gitpod.yml (hunk 1 above)
Comment (csweichel) [reply .gitpod.yml:15]: #13 removes the Makefile :)

File: .gitpod.yml (hunk 2 above)
Comment (rl-gitpod) [reply .gitpod.yml:17]: WFM for the last 2/3 days 🤷

File: .gitpod.yml (hunk 1 above)
Comment (rl-gitpod) [reply .gitpod.yml:15]: Hrm, I've been expanding the Makefile... go + Makefiles are a common expectation.

File: .gitpod.yml (hunk 2 above)
Comment (csweichel) [reply .gitpod.yml:17]: It does work - I stand corrected :) 

Now I'm left wondering why it works 🤷

**Pull request merged**